src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy

```diff
@@ -385,4 +389,36 @@ class BidderParamsSpec extends BaseSpec {
         assert bidderRequest.ext.prebid.server.datacenter == serverDataCenter
         assert bidderRequest.ext.prebid.server.gvlId == serverHostVendorId
     }
+
+    def "PBS should request to bidder with header Content-Encoding = gzip when adapters.BIDDER.endpointCompression = GZIP"() {
+        given: "PBS with adapter configuration"
+        def pbsService = pbsServiceFactory.getService(["adapters.generic.enabled"  : "true",
+                                                        "adapters.generic.endpointCompression"      : GZIP.name()])
+
+        and: "Default bid request"
+        def bidRequest = BidRequest.defaultBidRequest
+
+        when: "PBS processes auction request"
+        def response = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain header Content-Encoding = gzip"
+        assert response.ext?.debug?.httpcalls?.get(GENERIC.value)?.requestheaders?.first()
```

Comment:
Please use headers from Mock Server to make sure that you are passing the header.